src/build/plugins/server-entry-exports.ts

```diff
@@ -0,0 +1,28 @@
+import type { Nitro } from "nitro/types";
+import type { Plugin } from "rollup";
+
+export function serverEntryExports(nitro: Nitro): Plugin {
+  return {
+    name: "nitro:server-entry-exports",
+    transform(code, id) {
+      // Only append if serverEntry is defined
+      if (!nitro.options.serverEntry) {
+        return null;
+      }
+
+      // Only process the preset entry file (need to add .mjs extension)
+      const entryWithExt = nitro.options.entry + ".mjs";
+      if (id !== entryWithExt) {
+        return null;
+      }
+
+      // Append re-export statement before code is bundled
+      const reExport = `\nexport * from "#nitro-internal-virtual/server-entry";\n`;
```

Comment:
We need to create a new virtual that merges the main and user exports.

Then, we need to parse real `nitro.options.entry` and if there is a default export, also export it (otherwise, named exports and validate to avoid conflict)

`default` needs to be explicitly exported **if preset needs a `default` export**.


[this](https://github.com/unjs/obuild/blob/main/src/builders/transform.ts#L116-L129) might give you some ideas.